pytorch_lightning/profiler/pytorch/pytorch_kineto.py

```diff
@@ -0,0 +1,338 @@
+                self.profiler.__enter__()
+
+            if self._register is not None:
+                self._register.__enter__()
+
+        if self._lightning_module is not None:
+            # when the model is used in automatic optimization,
+            # we use `optimizer_step_and_closure` to step the model.
+            if self._lightning_module.automatic_optimization and "training_step" in self.STEP_FUNCTIONS:
+                self.STEP_FUNCTIONS.remove("training_step")
+
+        if (
+            self.profiler is not None
+            and (action_name in self._record_functions or action_name.startswith(self.RECORD_FUNCTION_PREFIX))
+            and action_name not in self._recording_map
+        ):
+            recording = record_function(action_name)
+            recording.__enter__()
+            self._recording_map[action_name] = recording
+            self._action_step_num[action_name] += 1
+
+    def stop(self, action_name: str) -> None:
+        if action_name in self._recording_map:
+            self._recording_map[action_name].__exit__(None, None, None)
+            del self._recording_map[action_name]
+
+        if not _KINETO_AVAILABLE or self._emit_nvtx:
+            return
+
+        if self.profiler is not None and (
+            action_name in self.STEP_FUNCTIONS or action_name.startswith(self.STEP_FUNCTION_PREFIX)
+        ):
+            step_num = self._action_step_num[action_name]
+            if step_num == self._wait_step:
+                # warm up
+                self._start_profiler()
+                self._stop_profiler()
+            elif step_num == self._wait_step + self._warmup_step:
+                # begin profile
+                self._start_profiler()
+            elif step_num == self._wait_step + self._warmup_step + self._active_step:
+                self._stop_profiler()
+                if self.dirpath is not None:
+                    self._save_result(action_name)
+                else:
+                    rank_zero_warn("The PyTorchProfiler failed to export trace as `dirpath` is None")
+
+    def summary(self) -> str:
+        if not self._profiler_kwargs.get("enabled", True) or self._emit_nvtx:
+            return ""
+
+        self._delete_profilers()
+
+        if not self.function_events:
+            return ""
+
+        if self._export_to_chrome and not _KINETO_AVAILABLE:
+            filename = f"{self.local_rank}_trace.json"
+            path_to_trace = filename if self.dirpath is None else os.path.join(self.dirpath, filename)
+            self.function_events.export_chrome_trace(path_to_trace)
+
+        data = self.function_events.key_averages(group_by_input_shapes=self._group_by_input_shapes)
+        table = data.table(sort_by=self._sort_by_key, row_limit=self._row_limit)
+
+        recorded_stats = {"records": table}
+        return self._stats_to_str(recorded_stats)
+
+    def _start_profiler(self) -> None:
+        if not _KINETO_AVAILABLE or self._emit_nvtx:
```

Comment:
Do we want to support this ?

Reply:
yeah, the nvtx is supported with current pytorch lightning profiler.